tests/acceptance/main-test.js

```diff
@@ -0,0 +1,42 @@
+import Ember from 'ember';
+import { module, test } from 'qunit';
+import startApp from '../../tests/helpers/start-app';
+import MainRoute from '../../tests/helpers/page-objects/main';
+
+const { run } = Ember;
+
+module('Acceptance | main', {
+  beforeEach: function() {
+    this.application = startApp();
+  },
+
+  afterEach: function() {
+    run(this.application, 'destroy');
+  }
+});
+
+test('it displays validation errors', (assert) => {
+  assert.expect(4);
+
+  return new MainRoute(assert, { routeName: '/' })
+    .assertVisitUrl()
+    .firstName('Milton')
```

Comment:
I think this function needs a verb. For example, are we filling in here or asserting?

Ok – I know that we're filling in by looking at the PageObject helper, but I find this naming very un-intuitive. I would prefer `fillInName('Milton')` or `fillInByLabel('name', 'Milton')`. Maybe I'm misunderstanding the PageObject library in general, but it seems that it is trying to do too much by trying to handle finders, interactions, and assertions. I would think that the library should simply take elements in the DOM and make them objects for our tests. You could persuade me to implement some `find` methods in there as well, but I definitely think assertions and interactions should be extracted.

I also think this test should ensure that the `firstName()` actually filled in the field. The `assertIsValid()` later is of little use if the field itself is not in the expected state.